tiled/server/router.py

```diff
@@ -779,21 +781,28 @@ async def put_metadata(
 
     metadata_modified = False
 
-    for spec in specs:
-        if spec in validation_registry:
-            try:
-                result = validation_registry(spec)(
-                    metadata, structure_family, structure, spec, references
-                )
-                if result is not None:
-                    metadata_modified = True
-                    metadata = result
-
-            except ValidationError as e:
-                raise HTTPException(
-                    status_code=400,
-                    detail=f"failed validation for spec {spec}:\n{e}",
-                )
+    # Specs should be ordered from most specific/constrained to least.
```

Comment:
Will it be obvious to the caller that if submit ["spec1", "spec2"] the result might be different from ["spec2", "spec1"]?

Reply:
Probably not obvious, but I can't think of a better alternative.

The idea that specs should go from specific to general has been in there from the beginning. The first practical consequence is that, if there is a special client (e.g. the `BlueskyRun` client object) for multiple specs the earliest spec in the list gets precedence.

The second practical consequence is that we have to choose _some_ order for the validators to run in.